components/esp_modem/src/esp_modem_c_api.cpp

```diff
@@ -84,6 +47,7 @@ extern "C" esp_modem_dce_t *esp_modem_new_dev(esp_modem_dce_device_t module, con
         delete dce_wrap;
         return nullptr;
     }
+    dce_wrap->dte = dte.get(); // Save plain pointer to DTE
```

Comment:
This is I think potentially dangerous, as we keep the plain pointer and two lines below we transfer its ownership to `dce`. Since we create a shared pointer from `create_uart_dte()` I think it better to make a copy.